packages/client-core/src/components/ReportUser/index.tsx

```diff
@@ -0,0 +1,350 @@
+import { twMerge } from 'tailwind-merge'
+import { NotificationService } from '../../common/services/NotificationService'
+import { LocationState } from '../../social/services/LocationService'
+import { ReportUserState } from '../../util/ReportUserState'
+import { uploadToFeathersService } from '../../util/upload'
+import { smallIconButtonStyles } from '../Glass/Buttons'
+import { useNavigationProvider } from '../Glass/NavigationProvider'
+
+const containerStyles = `
+  pointer-events-auto
+  inline-grid
+  absolute z-20
+  w-full
+  md:w-auto md:min-w-[40rem]
+  bottom-0 right-0 top-0
+  transition-transform
+`
+
+const sidebarContainerStyles = `
+  inline-grid
+  grid-rows-[min-content_min-content_1fr]
+  content-start
+  p-6
+  gap-y-8
+  max-h-full
+  h-full
+  border-l-2
+  border-white/10
+  shadow-[0_0.1rem_2.3rem_-0.5rem_hsla(0,0%,0%,0.1)]
+  backdrop-blur-3xl
+`
+
+const buttonContainer_base = `
+  absolute z-10
+  inline-flex items-center
+  -translate-y-1/2
+  lg:bottom-auto
+  lg:translate-y-0
+`
+
+const closeButtonStyles = `
+  right-4
+  text-white
+`
+
+const backButtonStyles = `
+  left-4
+`
+
+const actionButtonStyles = `
+  flex h-10 px-10 py-1 justify-center 
+  items-center self-stretch
+  text-white
+  rounded-full
+  bg-[rgba(255,255,255,0.2)]
+  shadow-[inset_0px_1px_1px_rgba(255,255,255,0.25),inset_0px_-1px_1px_rgba(255,255,255,0.1),0px_8px_6px_rgba(0,0,0,0.05)]
+`
+
+const inputDivStyles = `
+  flex flex-col
+  text-white
+`
+
+type ReportMenuProps = { type: ModerationTypeType }
+
+const REPORT_INPROGRESS = 'REPORT_INPROGRESS'
+const REPORT_SUCCESS = 'REPORT_SUCCESS'
+
+const ReportUserMenu = (props: ReportMenuProps) => {
+  const { t } = useTranslation()
+  const { type } = props
+  const { reportedPeerId, reportingUser } = useMutableState(ReportUserState)
+  const reportedUserId = type === 'user' ? NetworkState.mediaNetwork?.peers?.[reportedPeerId.value!]?.userId : undefined
+  const typeReport = type === 'location' ? 'Location' : 'User'
+  const currentLocation = getState(LocationState).currentLocation.location
+  const reportedLocationId = currentLocation.id
+  const userReportsMutation = useMutation(moderationPath)
+  const { activeHistoryKey, sidebarKey, navigateClose, navigateTo } = useNavigationProvider()
+
+  const [content, setContent] = useState<string>(REPORT_INPROGRESS) // REPORT_INPROGRESS | REPORT_SUCCESS
+  const fileInputRef = useRef<HTMLInputElement>(null)
+
+  const formData = useHookstate({
+    abuseType: 'null' as AbuseReasonsType,
+    details: '',
+    files: [] as File[]
+  })
+
+  const errors = useHookstate({
+    abuseType: '',
+    details: '',
+    files: ''
+  })
+
+  /** may need to move this to its own provider if used in more than several areas */
+  const fieldOptions = {
+    abuseType: {
+      label: t('user:usermenu.profile.typeAbuse'),
+      validate: (input: string) => {
+        if (input === null || input === '' || input === 'null') {
```

Comment:
🐑 Its normally worth trimming the input value before validating, otherwise an empty space/tab/return will pass this validation.

```suggestion
        input = input?.trim()
        if (input === null || input === '' || input === 'null') {
```